modules/apps/document-library-opener/document-library-opener-google-drive-web/src/main/java/com/liferay/document/library/opener/google/drive/web/internal/oauth/OAuth2Manager.java

```diff
@@ -243,13 +284,15 @@ protected void deactivate() {
 							dlGoogleDriveCompanyConfiguration.clientSecret()),
 						Collections.singleton(DriveScopes.DRIVE_FILE));
 
-			googleAuthorizationCodeFlowBuilder =
-				googleAuthorizationCodeFlowBuilder.setDataStoreFactory(
-					MemoryDataStoreFactory.getDefaultInstance());
-
```

Comment:
Why didn't you simply implement a new DataStoreFactory? I'm not 100% sure, but it looks like implementing a custom DataStore that uses your StoreCredentialStoreUtil in the impl. should be enough, and nothing else would need to be changed.

Reply:
I already though about this option, let me check why I discarded this option.